contents/data/sql/SQL.md

```diff
@@ -0,0 +1,241 @@
+</box>
+
+The example shown here is pretty typical. However, SQL queries for retrieving information have the capability to be much more complex as there can be a lot of other clauses, functions, and operators that are not covered here. As the complexity of the query grows, there is a possible decrease in the performance of the query. This is why [query planning](https://www.khanacademy.org/computing/computer-programming/sql/relational-queries-in-sql/a/more-efficient-sql-with-query-planning-and-optimization) and [query optimization](https://www.sisense.com/blog/8-ways-fine-tune-sql-queries-production-databases/) is important.
+
+---
+
+## Why learn SQL?
+
+Now that you have seen how SQL is used, let's take a look at some of the main reasons behind the widespread adoption of SQL and why you should learn it.
+
+### Easy to learn
+
+As you saw in the example above, SQL is not really that complex. Therefore, it is easy to learn, even for beginners who do not have any prior experience with databases. Since it has been around for a few decades, there are many good books and online resources to learn from.
+In addition, SQL and its distributions have a huge support community (such as [Stack Overflow](https://stackoverflow.com/questions/tagged/sql), and the [official MySQL forum](https://forums.mysql.com/)) which can prove useful when you run into problems while using SQL.
+
+### Free
+
+One of the benefits (and reasons) for SQL's popularity is that there are free distributions available (like MySQL, PostgreSQL and SQLite) as well as paid ones (like Microsoft SQL Server and Oracle) that come with more functionality.
+
+One such distribution is MySQL. It is free of cost and comes with [official](https://dev.mysql.com/doc/workbench/en/) as well third party easy-to-use <tooltip content="Graphical User Interface">GUIs</tooltip>, which are less daunting to new users as compared to a <tooltip content="Command Line Interface">CLI</tooltip>.
+
+### Highly Compatible
+
+SQL distributions work on many operating systems and more importantly, can be integrated with many languages. This includes languages like PHP, PERL, C++, Ruby, Java as well as *relatively "newer"* ones like Python, JavaScript (Node.js) and Go.
+
+In addition, online playgrounds like [DB Fiddle](https://www.db-fiddle.com/) and [JDoodle](https://www.jdoodle.com/execute-sql-online) make it easy to use and learn the SQL language quickly without the hassle of setting up an environment or application.
+
+## Disadvantages
+
+SQL is not without its problems. In general, the biggest problem is with regards to the features of SQL.
+
+Although SQL databases use established <tooltip content="American National Standard Institutes">ANSI</tooltip> & <tooltip content="International Organization for Standardization">ISO</tooltip> standards, some distributions (PostgreSQL, for example) add proprietary extensions to the standard SQL to ensure customer lock-in.
```

Comment:
Well, that's a point of view. The vendors might say the extensions are to give users extra value, not to ensure lock-in :-)